pkg/virtualization/core/storage/service/ImageManagementService.go

```diff
@@ -173,3 +173,57 @@ func (ims *ImageManagementService) CreateDisk(settings *disk.VirtualHardDiskSett
 	defer job.Close()
 	return job.WaitForJobCompletion(result.ReturnValue, -1)
 }
+
+func (ims *ImageManagementService) GetVirtualHardDiskConfig(path string) (size uint64,
+	blockSize uint32,
+	lSectorSize uint32,
+	pSectorSize uint32,
```

Comment:
It would be better to create some disk config type and return it (instead of these returns). In addition, above GetVirtualHardDiskSettingDataFromXml may return same type.

Reply:
I had considered adding that but since there are just 5 properties which need to extract from WMI it would be concise to return individual variables. This should be converted to a structure if more structure members need to copy out.

For reference, the properties which need to be copied as guided by what we can configure via create command:
`Command
    az stack-hci-vm disk create : Create a virtual hard disk.

Arguments
    --block-size-bytes             : Block size.
    --disk-file-format             : Disk file format of the virtual hard disk - vhd or vhdx.
        Usage: --disk-file-format "vhd".
    --logical-sector-bytes         : Logical Sector.
    --physical-sector-bytes        : Physical Sector.
    --size-gb                      : Size of the disk in GB.